src/main/java/info/smart_tools/smartactors/utils/reflection/conversion/ConversionWorker.java

```
@@ -0,0 +1,65 @@
+
+import info.smart_tools.smartactors.utils.reflection.ReflectionHelper;
+
+/**
+ * Type converter for conversion of intermediate types of IObject values to needed classes.
+ */
+public abstract class ConversionWorker {
+    private static final Logger logger = LogManager.getLogger(ConversionWorker.class);
+
+    /**
+     * Converts given Object to targetClass using {@code ConversionWorker} subclasses.
+     * The calling code must perform type casting by itself. Example of usage:
+     * <pre>
+     * {@code
+     * String strVal = "456";
+     * Integer intVal = (Integer) ConversionWorker.convert(Integer.class, strVal);
+     *
+     * ArrayList<String> list = new ArrayList<>();
+     * list.add("1");
+     * list.add("2");
+     * list.add("3");
+     * int[] array = (int[]) ConversionWorker.convert(int[].class, list);
+     *
+     * }
+     * </pre>
+     *
+     *
+     * @param targetClass is the class, which instance is needed as result
+     * @param input is given object
+     * @return instance of {@code targetClass}
+     * @throws IllegalArgumentException if conversion was not successful
+     */
+    public static Object convert(Class<?> targetClass, Object input) throws IllegalArgumentException {
+        if (targetClass == null) {
+            throw new IllegalArgumentException("Target class must be not null!");
+        }
+        if (input != null) {
+            if (targetClass.isAssignableFrom(input.getClass())) {
+                return input;
+            }
+            List<Class<? extends ConversionWorker>> workerClasses = ReflectionHelper.getSubClassesAsList(ConversionWorker.class);
+            for (Class<? extends ConversionWorker> workerClass : workerClasses) {
+                try {
+                    return workerClass.newInstance()
+                            .transform(targetClass, input);
+                } catch (IllegalArgumentException | InstantiationException | IllegalAccessException e) {
```

Why should we eat exceptions?